vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/IntegerField.java

```diff
@@ -0,0 +1,201 @@
+        if (valueFormClient == null || valueFormClient.isEmpty()) {
+            return null;
+        }
+        try {
+            return Integer.parseInt(valueFormClient);
+        } catch (NumberFormatException e) {
+            return null;
+        }
+    };
+
+    private static final SerializableFunction<Integer, String> FORMATTER = valueFromModel -> valueFromModel == null
+            ? ""
+            : valueFromModel.toString();
+
+    /**
+     * Constructs an empty {@code IntegerField}.
+     */
+    public IntegerField() {
+        super(PARSER, FORMATTER, Integer.MIN_VALUE, Integer.MAX_VALUE);
+    }
+
+    /**
+     * Constructs an empty {@code IntegerField} with the given label.
+     *
+     * @param label
+     *            the text to set as the label
+     */
+    public IntegerField(String label) {
+        this();
+        setLabel(label);
+    }
+
+    /**
+     * Constructs an empty {@code IntegerField} with the given label and
+     * placeholder text.
+     *
+     * @param label
+     *            the text to set as the label
+     * @param placeholder
+     *            the placeholder text to set
+     */
+    public IntegerField(String label, String placeholder) {
+        this(label);
+        setPlaceholder(placeholder);
+    }
+
+    /**
+     * Constructs an empty {@code IntegerField} with a value change listener.
+     *
+     * @param listener
+     *            the value change listener
+     *
+     * @see #addValueChangeListener(ValueChangeListener)
+     */
+    public IntegerField(
+            ValueChangeListener<? super ComponentValueChangeEvent<IntegerField, Integer>> listener) {
+        this();
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Constructs an empty {@code IntegerField} with a value change listener and
+     * a label.
+     *
+     * @param label
+     *            the text to set as the label
+     * @param listener
+     *            the value change listener
+     *
+     * @see #setLabel(String)
+     * @see #addValueChangeListener(ValueChangeListener)
+     */
+    public IntegerField(String label,
+            ValueChangeListener<? super ComponentValueChangeEvent<IntegerField, Integer>> listener) {
+        this(label);
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Constructs a {@code IntegerField} with a value change listener, a label
+     * and an initial value.
+     *
+     * @param label
+     *            the text to set as the label
+     * @param initialValue
+     *            the initial value
+     * @param listener
+     *            the value change listener
+     *
+     * @see #setLabel(String)
+     * @see #setValue(Object)
+     * @see #addValueChangeListener(ValueChangeListener)
+     */
+    public IntegerField(String label, Integer initialValue,
+            ValueChangeListener<? super ComponentValueChangeEvent<IntegerField, Integer>> listener) {
```

Comment:
I wouldn't like to add this many constructors, but this is consistent with `NumberField`.